### PR TITLE
PCX-2484: Add kotlin example checkout app

### PR DESCRIPTION
**Pull request by Fbada006**
More details in ticket

### CODE REVIEW

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutActivity.java [hunk 1]
```diff
@@ -7,6 +7,9 @@
  */
 package com.payoneer.checkout.examplecheckout;
 
+import static com.payoneer.checkout.examplecheckout.utils.ConstantsKt.CHARGE_PRESET_ACCOUNT_REQUEST_CODE;
```
Comment (siebeprojects) [inline example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutActivity.java:10]: This ExampleCheckoutActivity Java class should not reference any kotlin code. 
Lets keep the request ids inside the Activity

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import android.os.Bundle
+import android.util.Patterns
+import android.view.inputmethod.InputMethodManager
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.isVisible
+import androidx.test.espresso.IdlingResource
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.payoneer.checkout.examplecheckout.databinding.ActivityExamplecheckoutBinding
+import com.payoneer.checkout.examplecheckout.utils.CHARGE_PRESET_ACCOUNT_REQUEST_CODE
```
Comment (siebeprojects) [inline example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt:23]: Can we declare the request ids inside this Activity, else we end up with some extra Kotlin classes but not for Java.

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/utils/Extensions.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ *  See the LICENSE file for more information.
+ *
+ */
+
+package com.payoneer.checkout.examplecheckout.utils
+
+import android.widget.TextView
+import com.payoneer.checkout.examplecheckout.R
+
+fun TextView.setLabel(message: String) {
```
Comment (siebeprojects) [inline example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/utils/Extensions.kt:16]: Similar to the Java Activity, can we keep this method inside the Kotlin Activity as well?

File: example-checkout/build.gradle [hunk 4]
```diff
@@ -70,12 +71,17 @@ android {
             }
         }
     }
+
+    buildFeatures {
```
Comment (siebeprojects) [inline example-checkout/build.gradle:75]: Can we also use View Binding in the Java Activity class? Then both Kotlin and Java example Activities are implemented the same.

File: example-checkout/src/androidTest/java/com/payoneer/checkout/examplecheckout/kotlin/CardPaymentTests.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+import com.payoneer.checkout.sharedtest.service.ListSettings;
+
+import androidx.test.espresso.IdlingResource;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+import androidx.test.rule.ActivityTestRule;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public final class CardPaymentTests extends KotlinAbstractTest {
```
Comment (siebeprojects) [inline example-checkout/src/androidTest/java/com/payoneer/checkout/examplecheckout/kotlin/CardPaymentTests.java:34]: Lets not duplicate all tests for Kotlin, instead let's do the following:

- make Kotlin the default Activity
- run all tests against the Kotlin Activity
- Create a new CheckoutJavaTests class with two tests:

1. Successful Visa payment
2. successful PayPal redirect payment.

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+import com.payoneer.checkout.examplecheckout.utils.setLabel
+import com.payoneer.checkout.ui.PaymentActivityResult
+import com.payoneer.checkout.ui.PaymentTheme
+import com.payoneer.checkout.ui.PaymentUI
+import com.payoneer.checkout.ui.page.idlingresource.SimpleIdlingResource
+
+/**
+ * This is the main Activity of this example app demonstrating how to use the Checkout SDK in kotlin
+ */
+class ExampleCheckoutKotlinActivity : AppCompatActivity() {
```
Comment (siebeprojects) [inline example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt:34]: Let's rename the ExampleCheckoutActivity -> ExampleCheckoutJavaActivity 
so it is similar to this Activity class name.

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutActivity.java (hunk 1 above)
Comment (Fbada006) [reply example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutActivity.java:10]: Done

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt (hunk 2 above)
Comment (Fbada006) [reply example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt:23]: Alright, let me put them in companion objects

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/utils/Extensions.kt (hunk 3 above)
Comment (Fbada006) [reply example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/utils/Extensions.kt]: This has been done

File: example-checkout/build.gradle (hunk 4 above)
Comment (Fbada006) [reply example-checkout/build.gradle:75]: Yes we can and. It has been implemented

File: example-checkout/src/androidTest/java/com/payoneer/checkout/examplecheckout/kotlin/CardPaymentTests.java (hunk 5 above)
Comment (Fbada006) [reply example-checkout/src/androidTest/java/com/payoneer/checkout/examplecheckout/kotlin/CardPaymentTests.java]: Good point. The cleanup has been sorted out

File: example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt (hunk 6 above)
Comment (Fbada006) [reply example-checkout/src/main/java/com/payoneer/checkout/examplecheckout/ExampleCheckoutKotlinActivity.kt:32]: Done

**Review by siebeprojects — APPROVED** [review]

**Pull request merged**